src/main/webapp/search-meal-script.js

```diff
@@ -17,15 +17,20 @@ function searchMeal() {
     {
         const container = document.getElementById("dishes-container");
         container.innerText = "";
-        dishes = dishes ?? [""];
-        dishes.forEach((dish) => {
-            container.appendChild(createListElement(dish));
+        dishes = dishes ?? {0: ""};
+        Object.entries(dishes).forEach((dish) => {
+            container.appendChild(createListElement(dish[1]));
         });
     });
 }
 
-function createListElement(val) {
+function createListElement(obj) {
```

Comment:
You can name the argument "dish" so that it's easier to assume what kind of object is this.